static/js/tab_bar.js

```diff
@@ -78,18 +78,13 @@ function build_tab_bar(filter) {
         });
 
         $("#tab_list span:nth-last-child(2)").on("click", function (e) {
-            exports.open_search_bar_and_close_narrow_description();
-            search.initiate_search();
-            e.preventDefault();
-            e.stopPropagation();
-        });
-
-        // Hacky way of protecting the behaviour of links via preventDefault
-        // and stopPropagation
-        $(".narrow_description > a").on("click", function (e) {
-            window.location.href = e.target.href;
-            e.preventDefault();
-            e.stopPropagation();
+            // Let links behave normally, ie, do nothing if <a>
+            if (e.target.tagName.toLowerCase() !== 'a') {
```

Comment:
I think `if e.closest('a')` would be more accurate, since the link could be around the element that's actually clicked

Reply:
I wonder if `if (e.target.tagName.toLowerCase() === 'span') {` is even better, because then you can click on a code block and highlight a part of it, or select the text inside better, without the search bar triggering.

Reply:
I don't think span makes sense.  You should be able to highlight code blocks in any case; that's not a click event, it's a select event.  I added the missing conditional that handles that issue correctly.